File: Data Structures & Algorithms/daily-temperatures/submission-3.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:

        stack = []
        size = len(temperatures)
        res= [0 for i in range(size)]
        
        for i in range(size):
            while stack and temperatures[i] > temperatures[stack[-1]]:
                temp = stack.pop()
                res[temp] = i-temp
            stack.append(i)
        


        
        return res
        

File: Data Structures & Algorithms/binary-tree-postorder-traversal/submission-3.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    /**
     * @param {TreeNode} root
     * @return {number[]}
     */
    postorderTraversal(root) {

        const stack = [root];
        const res = [];

        if(!root){
            return res;
        }

        while (stack.length > 0){
            const node = stack.pop();
            res.push(node.val);
            if (node.left){
                stack.push(node.left);
            }

            if (node.right){
                stack.push(node.right);
            }

        }

        return res.reverse();
            
    }
}
